provider/ecs/ecs.go

```diff
@@ -200,103 +202,128 @@ func (p *Provider) loadECSConfig(ctx context.Context, client *awsClient) (*types
 // and the EC2 instance data
 func (p *Provider) listInstances(ctx context.Context, client *awsClient) ([]ecsInstance, error) {
 	var taskArns []*string
-	req, _ := client.ecs.ListTasksRequest(&ecs.ListTasksInput{
-		Cluster:       &p.Cluster,
-		DesiredStatus: aws.String(ecs.DesiredStatusRunning),
-	})
+	var instances []ecsInstance
+	var clustersArn []*string
+	var clusters Clusters
 
-	for ; req != nil; req = req.NextPage() {
-		if err := wrapAws(ctx, req); err != nil {
-			return nil, err
-		}
+	if p.AutoDiscoverClusters {
+		input := &ecs.ListClustersInput{}
+		result, _ := client.ecs.ListClusters(input)
+		for result.NextToken != nil {
+			clustersArn = append(clustersArn, result.ClusterArns...)
 
-		taskArns = append(taskArns, req.Data.(*ecs.ListTasksOutput).TaskArns...)
+			input.NextToken = result.NextToken
+			result, _ = client.ecs.ListClusters(input)
+		}
+		clustersArn = append(clustersArn, result.ClusterArns...)
```

Comment:
In the way to avoid repetition and possible problem if `result == nil`, what is your mind about this suggestion : 
```go
for {
    if result, _ := client.ecs.ListClusters(input); result != nil {
        clustersArn = append(clustersArn, result.ClusterArns...) 		  		 
	input.NextToken = result.NextToken
        if result.NextToken == nil {
            break
        }
    } else {
        break
    }
}